### ISSUE TITLE
MediaFailed event raises exception

### ISSUE THREAD

**Issue by taoyouh** [kakone/VLC.MediaElement#60]
XAML code: 

```
<vlc:MediaElement x:Name="mediaElement" Source="http://127.0.0.1" MediaFailed="MediaElement_MediaFailed"/>
```

Exception:

> {System.Exception: 应用程序调用一个已为另一线程整理的接口。 (Exception from HRESULT: 0x8001010E (RPC_E_WRONG_THREAD))
>    at Windows.UI.Xaml.RoutedEventArgs..ctor()
>    at VLC.MediaElement.<OnErrorAsync>d__180.MoveNext()}

It seems that RoutedEventArgs constructor should be called in UI thread.

**Comment by taoyouh** [comment]
The direct solution is to use dispatcher to call the constructors, which I am working on. However I think these event should not use `RoutedEventArgs` because they are not routed at all. Even if the event is routed, `RoutedEventHandler `should be used instead of `EventHandler`.

**Comment by kakone** [comment]
You are right, I'm going to rewrite these events to do some true routed events.

**Comment by kakone** [comment]
Resolved by 211399e92fe2533a0081cd72601affd6b09c61cb
